### DISCUSSION TITLE
control surface with 74hc165

### DISCUSSION THREAD

**Discussion by PanJabuszko** [tttapa/Control-Surface#913] [General]
Hi, I have an Arduino Uno and 24 keys foot keyboard in my room. I want to use 3 74hc165 shift registers in that project, but I don't have any idea how to declare in control surface a note to a binary number (00000001, 00000010, ...). 
I have managed to send button state here is the code:
```cpp
// PL pin 1
int load = 7;
// CE pin 15
int clockEnablePin = 4;
// Q7 pin 7
int dataIn = 5;
// CP pin 2
int clockIn = 6;

void setup(){
  
  Serial.begin(9600);
 
  pinMode(load, OUTPUT);
  pinMode(clockEnablePin, OUTPUT);
  pinMode(clockIn, OUTPUT);
  pinMode(dataIn, INPUT);
}
 
void loop(){
 
  digitalWrite(load, LOW);
  delayMicroseconds(5);
  digitalWrite(load, HIGH);
  delayMicroseconds(5);
 
  digitalWrite(clockIn, HIGH);
  digitalWrite(clockEnablePin, LOW);
  byte incoming = shiftIn(dataIn, clockIn, LSBFIRST);
  digitalWrite(clockEnablePin, HIGH);
 
  Serial.println(incoming, BIN);
  delay(200);
}
```
How to assign incoming to a midi?

I am counting for your help :DD

**Comment by tttapa** [comment]
You can use the `bitRead` function to get the state of each key in a `for` loop, then based on the state of the key, send a MIDI message as shown in https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html#midi_md-sending.
